osu.Game.Tests/Visual/Gameplay/TestSceneUnstableRateCounter.cs

```diff
@@ -0,0 +1,122 @@
+// Copyright (c) ppy Pty Ltd <contact@ppy.sh>. Licensed under the MIT Licence.
+// See the LICENCE file in the repository root for full licence text.
+
+using System;
+using System.Collections.Generic;
+using System.Diagnostics.CodeAnalysis;
+using System.Linq;
+using NUnit.Framework;
+using osu.Framework.Allocation;
+using osu.Framework.Graphics;
+using osu.Framework.Testing;
+using osu.Framework.Utils;
+using osu.Game.Rulesets.Judgements;
+using osu.Game.Rulesets.Osu.Objects;
+using osu.Game.Rulesets.Osu.Scoring;
+using osu.Game.Rulesets.Scoring;
+using osu.Game.Scoring;
+using osu.Game.Screens.Play;
+using osu.Game.Screens.Play.HUD;
+using osuTK;
+
+namespace osu.Game.Tests.Visual.Gameplay
+{
+    public class TestSceneUnstableRateCounter : OsuTestScene
+    {
+        [Cached(typeof(ScoreProcessor))]
+        private TestScoreProcessor scoreProcessor = new TestScoreProcessor();
+
+        [Cached(typeof(GameplayState))]
+        private GameplayState gameplayState;
+
+        private OsuHitWindows hitWindows = new OsuHitWindows();
+
+        private double prev;
+
+        public TestSceneUnstableRateCounter()
+        {
+            Score score = new Score
+            {
+                ScoreInfo = new ScoreInfo(),
+            };
+            gameplayState = new GameplayState(null, null, null, score);
+            scoreProcessor.NewJudgement += result => scoreProcessor.PopulateScore(score.ScoreInfo);
+        }
+
+        [SetUpSteps]
+        public void SetUp()
+        {
+            AddStep("Reset Score Processor", () => scoreProcessor.Reset());
+        }
+
+        [Test]
+        public void TestBasic()
+        {
+            AddStep("Create Display", () => recreateDisplay());
+
+            AddRepeatStep("Set UR to 250", () => setUR(25), 20);
+
+            AddStep("Reset UR", () =>
+            {
+                scoreProcessor.Reset();
+                recreateDisplay();
+            });
+
+            AddRepeatStep("Set UR to 100", () => setUR(10), 20);
```

Comment:
100 or 10? Also running these tests in succession causes the values to bounce around the expectation, rather than stay at a solid number. Is that intended?

Reply:
The method was not accurately named before, I have refactored it to applyJudgement and added comments to explain it better.

The jumping around is intended. The step works by alternating +10ms and -10ms inputs, which even out to 100 UR, but if there's an unequal amount of +10ms then the -10ms (occurs on odd step numbers), then the UR different from 100.